library/src/scala/Tuple.scala

```diff
@@ -11,6 +11,10 @@ sealed trait Tuple extends Product {
   inline def toArray: Array[Object] =
     scala.runtime.Tuple.toArray(this)
 
+  /** Create a copy of this tuple as a List */
+  inline def toList: List[Object] =
```

Comment:
Returning a `List[Object]` will not be useful most of the time. For example if we have a `(1, 2, 3)` we would want a `List[Int]`. Or if we have `(String, Int)` we would probably want `List[String | Int]` instead of `List[Any]`.

This seems to be feasible with a match type like 
```scala
type Union[T <: Tuple] = T match
  case EmptyTuple => Nothing
  case h *: t => h | Union[t]
  case Tuple => Any

  inline def toList: List[Union[this.type]] = ...
```

Reply:
We already have a `Map` type match on `Tuple`, it looks like a fold operation might also be useful here. Something like:

```scala
type Fold[Z, D, F[_, _], T <: Tuple] = T match
  case EmptyTuple => Z
  case h *: t => F[h, Fold[Z, D, t]]
  case Tuple => D

type Union[T <: Tuple] = Fold[Nothing, Any, _ | _, T]
```

Reply:
Also see #9212

Reply:
I'm adding the `Fold` in #9217

Reply:
See https://github.com/lampepfl/dotty/pull/9217/files#diff-77f62e2e20dc3fb26d827b07b376a861R8